documentation/Database-Schema-Application.md

```diff
@@ -0,0 +1,56 @@
+   * An application with Draft should be flagged as `is_active`?
+
+* <b>application_section</b>
+Sections of the application that contain some elements. This is exactly the same as the `template_section`.
+The `application_id` links to the application.
+The `template_section_id` links to the template.
+<b>To be considered: Why so we need an `application_section` and not `application_element`. We should use the same logic for both.</b>
+
+* <b>application_stage_history</b>
+There is one or more stages per application template. Each one is defined by `template_stage`. The application stage history will keep records for when the application changed to each particular stage (previously defined for this application template).
+The `application_id` links to the application.
+The `stage_id` links to the template stage, which will have the `name` and number of this stage.
+The `time_created` defined when this stage was created.
+The `is_current` is set to `'true'` while this is the current stage of this application. Otherwise is set to `'false'`.
+<b>To be considered: When the application review is not in progress anymore do we keep the last `is_active` stage as `'true'`?</b>
+
+* <b>application_status_history</b>
+For each stage of the application it can have associated statuses. The application status history will keep records for when the application changed to each particular status of a stage. 
+The `application_stage_history_id` links to the application stage (and the application itself).
+The `status` is either `'Draft'`, `'Withdrawn'`, `'Submitted'`, `'Changes Required'`, `'Re-submitted'`, `'Completed'` and self-explanatory,
+The `time_created` defined when this status of a stage was created.
+The `is_current` is set to `'true'` while this is the current status of this application stage. Otherwise is set to `'false'`.
+
+### Question response
+For every new answer that the Applicant give to each resquested elements (questions) in the application a new <b>application response</b> is created. The Applicant user can make changes to responses until submitting or re-submitting the application to be reviewed. The 
+
+* <b>application_reponse</b>
+The `template_question_id` links to the template element, which defines what is the question and how to render in the application.
+The `application_id` links to the application.
+The `value` is what the Applicant user enter as a response.
+The `time_created` defined when this response was created.
+More detailed description of response coming soon: `value`.
+<b>To be considered:</b>
+  * Should we change `template_question_id` to `template_element_id` to keep consistent with the table name?
+  * How can we flag a question to be editable by the user when Changes are required if it's only defined per template in `template_element`? Considering what was also flagged in the `application_section` we might need another `application_question` table to deal with that.
```

Comment:
Wouldn't this be defined by `review_response.review_decision`? If that value is "Observations" (and the associated Review has been submitted by reviewer), then this question become editable by Applicant.

Reply:
Yes. As discussed in today's meeting. We would be using a SQL view or function to deduce this from the review decisions. 👍